standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/cache/CachedStore.java

```diff
@@ -1702,29 +2030,58 @@ private ColumnStatistics adjustColStatForGet(Map<String, String> tableParams,
     return colStat;
   }
 
+  private static void updateTableColumnsStatsInternal(Configuration conf, ColumnStatistics colStats,
+                                                      Map<String, String> newParams, String validWriteIds,
+                                                      long writeId) throws MetaException {
+    String catName = colStats.getStatsDesc().isSetCatName() ?
+            normalizeIdentifier(colStats.getStatsDesc().getCatName()) :
+            getDefaultCatalog(conf);
+    String dbName = normalizeIdentifier(colStats.getStatsDesc().getDbName());
+    String tblName = normalizeIdentifier(colStats.getStatsDesc().getTableName());
+    if (!shouldCacheTable(catName, dbName, tblName)) {
+      return;
+    }
+    Table table = sharedCache.getTableFromCache(catName, dbName, tblName);
+    if (table == null) {
+      // The table is not yet loaded in cache
+      return;
+    }
+
+    boolean isTxn = TxnUtils.isTransactionalTable(table.getParameters());
```

Comment:
What's the difference between updatePartitionColumnStatistics and updateTableColumnStatistics? Why updateTableColumnStatistics need to check validWriteIds and updatePartitionColumnStatistics not? In ObjectStore, both do similar check.

Reply:
currently table col stats is fetched from cache but for partition col stats, its always fetched from rawstore. I think thats why txn stats are not handled for partitions col stats.

Reply:
I see, we need to fill this up later.